rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -305,6 +305,7 @@ public void addEmbeddedAttachments(
             HalResource halResource,
             Set<Attachment> attachments) {
         for (Attachment attachment : attachments) {
+            if(attachment == null)continue;
```

Comment:
A more convenient formatting would be:
```
if (attachment == null) {
    continue;
}
```

Reply:
And btw: how did you come across the situation that the list of attachments contains a null value? Is that possible in real life code or just in the documentation test?

Reply:
Yes @imaykay,  this check is needed only during documentation test.I have not encountered the scenario in real time.

Reply:
And do you think it would be possible to write the doc test in a way that this cannot happen here as well? I am not a big fan of production code that only exists because of test cases.

Reply:
Yes @imaykay . I have added a mock service call in test layer to generate an attachment object.